### PR TITLE
Feature/user picture

### PR DESCRIPTION
**Pull request by tkgs16**

### CODE REVIEW

File: src/app/_components/home/ProfilePicture/ProfilePictureDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      <DialogContent>
+        <ImageUpload
+          onUploadSuccess={() => {
+            onUploadSuccess();
+            toast({
+              title: "Success",
+              description: "Image uploaded successfully",
+            });
+            setIsOpen(false);
+          }}
```
Comment (Dolevkle) [inline src/app/_components/home/ProfilePicture/ProfilePictureDialog.tsx:37]: move it to upper function called `handleUploadSuccess` and then it'll be in the html
`<ImageUpload onUploadSuccess={handleUploadSuccess}/>`

File: src/app/_components/home/ProfilePicture/ProfilePictureDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    ? usersImages.map((image) => image.url)
+    : ["https://utfs.io/f/9dc01f95-3ff0-484f-822e-59500be4baeb-x8m314.png"];
+
+  const handleDeleteSuccess = async () => {
+    await refetchImages();
+  };
+
+  const handleUploadSuccess = async () => {
+    await refetchImages();
+  };
```
Comment (Dolevkle) [inline src/app/_components/home/ProfilePicture/ProfilePictureDisplay.tsx:36]:
```suggestion
  const handleUploadSuccess = async () => await refetchImages();
```

File: src/app/_components/loading-button.tsx
Comment (Dolevkle) [inline src/app/_components/loading-button.tsx:1]: there is already button component why also loading button.
you can do ```
    <Button disabled>
      <Loader2 className="mr-2 h-4 w-4 animate-spin" />
      Please wait
    </Button>
```

File: src/app/actions/imageRemove.ts
Comment (Dolevkle) [inline src/app/actions/imageRemove.ts:1]: format the action maybe call function `removeImageAction` and handle the error

File: src/server/api/routers/image/image.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+//     findUserById: protectedProcedure
+//     .input(z.string())
+//     .query(async ({ ctx, input }) => {
+//       return ctx.db.user.findUnique({
+//         where: {
+//           id: input,
+//         },
+//       });
+//     }),
+// });
```
Comment (Dolevkle) [inline src/server/api/routers/image/image.ts:38]: delete

File: src/app/_components/home/ProfilePicture/DeleteUserPicture.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        description: "Uh oh! Something went wrong.",
+      });
+      setLoading(false);
+    },
+  });
+
+  const handleRemove = async () => {
+    setLoading(true); // Start loading
+    const imageKey = imageUrl.substring(imageUrl.lastIndexOf("/") + 1);
+    const res = await imageRemove(imageKey);
```
[the remark this answers is not in the document]
Comment (tkgs16) [reply src/app/_components/home/ProfilePicture/DeleteUserPicture.tsx:41]: אשנה תשם, הimage remove זה השם שההודי ההוא שהראתי לך נתן ולא שמתי לב חחחחחחחחחחח
זה מה שמוחק מהS3
delete image זה מה שמוחק את התמונה מהדיבי

File: src/app/_components/home/ProfilePicture/ProfilePictureDialog.tsx (hunk 1 above)
Comment (tkgs16) [reply src/app/_components/home/ProfilePicture/ProfilePictureDialog.tsx:37]: done

File: src/app/_components/home/ProfilePicture/ProfilePictureDisplay.tsx (hunk 2 above)
Comment (tkgs16) [reply src/app/_components/home/ProfilePicture/ProfilePictureDisplay.tsx:36]: done

File: src/server/api/routers/image/image.ts (hunk 3 above)
Comment (tkgs16) [reply src/server/api/routers/image/image.ts:38]: done mb

File: src/app/_components/loading-button.tsx
Comment (tkgs16) [reply src/app/_components/loading-button.tsx:1]: actually i didnt use it in the end after i discovered what you wrote, so ill just remove this component no further changes needed

File: src/app/actions/imageRemove.ts
Comment (tkgs16) [reply src/app/actions/imageRemove.ts:1]: changed it

**Pull request merged**